### PR TITLE
Initial draft of upstream docs

### PR DESCRIPTION
**Pull request by tomgeorge**
This is the initial configuration/installation instructions.  Building/testing to come.

Signed-off-by: Tom George <tgeorge@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,77 @@
...
-| `IMAGE`                           | Name of image used for main pod |
-| `IMAGE_TAG`                       | Tag of image used for main pod |
-
-Finally, a secret containing the pod's serviceaccount's secret and id should be created with the data
-
-| Key | Value |
-| --- | ----- |
-| `service.account.secret` | Service account token |
-| `service.account.id` | User id for service account |
+Configuration is done via env vars pulled from `./deploy/helm/templates/configmap.yaml`, or `./deploy/openshift/configmap.yaml`, depending on the deployment method.
```
Comment (amisevsk) [inline README.md]:
```suggestion
Configuration is done via environment variables pulled from `./deploy/helm/templates/configmap.yaml`, or `./deploy/openshift/configmap.yaml`, depending on the deployment method.
```

File: README.md [hunk 2]
```diff
@@ -1,48 +1,77 @@
...
+The config values to be set are:
+
+| Env Var | Usage | Default |
+| -- | -- | -- |
+| `CACHING_INTERVAL_HOURS` | Interval, in hours, between checking health of daemonsets | `"1"` |
+| `CACHING_MEMORY_REQUEST` | The container memory request | `10Mi` |
+| `CACHING_MEMORY_LIMIT` | The container memory limit | `20Mi` |
```
Comment (amisevsk) [inline README.md]: Maybe something like "memory request(limit) for each cached image"

File: README.md [hunk 3]
```diff
@@ -1,48 +1,77 @@
...
+| `DAEMONSET_NAME`         | Name of daemonset to be created | `kubernetes-image-puller` |
+| `NAMESPACE`              | Namespace where daemonset is to be created. Shared for all users | `k8s-image-puller` |
```
Comment (amisevsk) [inline README.md]: "Shared for all users" only really makes sense downstream (where users ~ clusters)

File: README.md [hunk 4]
```diff
@@ -1,48 +1,77 @@
...
+| `IMAGES`                 | List of images to be cached, in the format `<name>=<image>;...` | Contains a default list of images, but should be configured when deploying |
+| `NODE_SELECTOR` | The node selector to run the daemonsets on particular nodes. | `'{}'` |
```
Comment (amisevsk) [inline README.md]:
```suggestion
| `NODE_SELECTOR` | Node selector applied to pods created by the daemonset. | `'{}'` |
```

File: README.md [hunk 5]
```diff
@@ -1,48 +1,77 @@
...
-The application is meant to be used in conjunction with [fabric-oso-proxy](https://github.com/fabric8-services/fabric8-oso-proxy) and [fabric8-auth](https://github.com/fabric8-services/fabric8-auth).
+This is an upstream version of the [kubernetes-image-puller](https://github.com/redhat-developer/kubernetes-image-puller).  Where the downstream puller requires integrations with [fabric-oso-proxy](https://github.com/fabric8-services/fabric8-oso-proxy) and [fabric8-auth](https://github.com/fabric8-services/fabric8-auth), and impersonates users in multiple clusters, this application is meant to run on a single cluster, and pre-pull Eclipse Che images.
 
-Fabric8-auth is used to obtain a service account token, which is used along with an impersonate header to create resources in other clusters/namespaces as directed by fabric8-oso-proxy.
+To cache images, this app creates a Daemonset on the desired cluster, which in turn creates a pod on each node in the cluster consisting of a list of containers with command `sleep infinity`. This ensures that all nodes in the cluster have those images cached. We also periodically check the health of the daemonset and re-create it if necessary.
```
Comment (amisevsk) [inline README.md]: Technically speaking, `sleep 30d` (which we do) is more compatible than `sleep infinity`. IIRC, `sleep` in busybox doesn't support "infinity".

File: README.md (hunk 4 above)
Comment (tomgeorge) [reply README.md]: Thank you, I wasn't 100% sure what this did :D

**Comment by tomgeorge** [comment]
Updated

**Pull request merged**